### PR TITLE
drop skip_register_redirect + some refactoring

### LINKED ISSUE #12417: drop 'skip_register_redirect' in fxa configs
to quote myself from #10487 
> after the authentication has taken place the only difference is whether we redirect users to the edit profile page... something I'm not sure we need anyway because it was originally implemented when addons-frontend was mobile only and we'd not gotten around to implementing the edit page. 🤔 

(splitting this out as an issue so we're clear on the feature removal)

The logic currently is:
- `default` fxa config (used everywhere on django served pages, and possibly frontend*) redirects for a new user everywhere (locally, addons-dev, addons-stage, addons-prod)
- `amo` fxa config (used by frontend*) redirects locally & addons-dev, but doesn't on addons-stage & addons-prod
- `local` fxa config (used for frontend local dev) redirects locally & addons-dev & addons-stage (it isn't available on addons-prod)
- `code-manager` fxa config (used for code manager app) copies the `amo` logic.

IMO, we should either stop redirecting any new user to the profile registration page, or redirect all new users.  The reason it was implemented (not having a frontend edit profile page) is long gone and it adds complexity on the server side (and means as many as 4 times as many fxa client key pairs need to be generated per environment)

@jvillalobos to decide if we want to redirect users who are first time AMO users to the account profile settings page or not.

*frontend uses `amo` on some environments and possibly not others but it's an ops env setting so a little opaque.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #12417

### CODE REVIEW

File: src/olympia/conf/prod/settings.py [hunk 1]
```diff
@@ -85,34 +85,22 @@
...
     'amo': {
         'client_id': env('AMO_FXA_CLIENT_ID'),
         'client_secret': env('AMO_FXA_CLIENT_SECRET'),
-        'content_host': 'https://accounts.firefox.com',
-        'oauth_host': 'https://oauth.accounts.firefox.com/v1',
-        'profile_host': 'https://profile.accounts.firefox.com/v1',
-        'redirect_url': 'https://addons.mozilla.org/api/v3/accounts/authenticate/?config=amo', # noqa
-        'scope': 'profile',
-        'skip_register_redirect': True,
+        # redirect_url = 'https://addons.mozilla.org/api/v3/accounts/authenticate/?config=amo', # noqa
```
Comment (EnTeQuAk) [inline src/olympia/conf/prod/settings.py]: Hmm, if you're commenting this one out in all the configs it's probably best to add an additional comment why it's commented out. Are you using this for documentation reasons?

Comment (eviljeff) [reply src/olympia/conf/prod/settings.py]: There's no way of knowing what the redirect url is going to if you didn't personally create the fxa client, so I'm documenting it, yes.

Comment (EnTeQuAk) [reply src/olympia/conf/prod/settings.py]: Okay, then it'd make sense to add some more context as part of that documentation, e.g the purpose of that line and why it's here etc.

Comment (eviljeff) [reply src/olympia/conf/prod/settings.py]: https://github.com/mozilla/addons-server/pull/12532/commits/8a5b67d43e4b41d1ffaf73648c01d0a2ddb58367

File: src/olympia/lib/settings_base.py [hunk 2]
```diff
@@ -1782,6 +1782,9 @@ def get_raven_release():
     ('olympia', 'amo'),
 )
 
+FXA_CONTENT_HOST = 'https://stable.dev.lcip.org'
+FXA_OAUTH_HOST = 'https://oauth-stable.dev.lcip.org/v1'
+FXA_PROFILE_HOST = 'https://stable.dev.lcip.org/profile/v1'
```
Comment (diox) [inline src/olympia/lib/settings_base.py]: Maybe the default should be prod instead, and these reset in `dev/settings.py` and `settings.py` ? We haven't been super consistent with this.

In any case stage should use prod settings for these - that needs to be fixed.

Comment (diox) [reply src/olympia/lib/settings_base.py]: While you're at it, can you get rid of `VALID_LOGIN_REDIRECTS` ? It's used in `devhub/views.py` in `_clean_next_url()` but judging from the contents of that variable it's completely outdated, gotta be obsolete.

**Pull request merged**